src/manifest/__tests__/manifest.test.ts

```diff
@@ -0,0 +1,641 @@
+
+    const manifest = new Manifest(simpleFakeManifest, {});
+    expect(manifest.isLive).toEqual(true);
+    expect(manifest.duration).toEqual(undefined);
+
+    expect(logSpy).not.toHaveBeenCalled();
+    logSpy.mockRestore();
+  });
+
+  it("should correctly parse every manifest informations given", () => {
+    const oldPeriod1 = { id: "0" };
+    const oldPeriod2 = { id: "1" };
+    const oldManifestArgs = {
+      availabilityStartTime: 5,
+      baseURL: "test",
+      duration: 12,
+      id: "man",
+      isLive: false,
+      lifetime: 13,
+      minimumTime: 4,
+      parsingErrors: ["a", "b"],
+      periods: [oldPeriod1, oldPeriod2],
+      presentationLiveGap: 18,
+      suggestedPresentationDelay: 99,
+      timeShiftBufferDepth: 2,
+      transportType: "foobar",
+      uris: ["url1", "url2"],
+    };
+
+    const logSpy = jest.spyOn(log, "warn").mockImplementation(jest.fn());
+    const fakePeriod = jest.fn((period) => {
+      return { ...period, id: "foo" + period.id, parsingErrors: [period.id] };
+    });
+    jest.mock("../period", () =>  ({ default: fakePeriod }));
+    const Manifest = require("../manifest").default;
+    const manifest = new Manifest(oldManifestArgs, {});
+
+    expect(manifest.adaptations).toEqual({});
+    expect(manifest.availabilityStartTime).toEqual(5);
+    expect(manifest.baseURL).toEqual("test");
+    expect(manifest.getDuration()).toEqual(12);
+    expect(manifest.id).toEqual("man");
+    expect(manifest.isLive).toEqual(false);
+    expect(manifest.lifetime).toEqual(13);
+    expect(manifest.minimumTime).toEqual(4);
+    expect(manifest.parsingErrors).toEqual(["0", "1"]);
+    expect(manifest.periods).toEqual([
+      { id: "foo0", parsingErrors: ["0"] },
+      { id: "foo1", parsingErrors: ["1"] },
+    ]);
+    expect(manifest.presentationLiveGap).toEqual(18);
+    expect(manifest.suggestedPresentationDelay).toEqual(99);
+    expect(manifest.timeShiftBufferDepth).toEqual(2);
+    expect(manifest.transport).toEqual("foobar");
+    expect(manifest.uris).toEqual(["url1", "url2"]);
+    expect(logSpy).not.toHaveBeenCalled();
+    logSpy.mockRestore();
+  });
+
+  it("should update with a new Manifest when calling `update`", () => {
+    const oldManifestArgs = {
+      availabilityStartTime: 5,
+      baseURL: "test",
+      duration: 12,
+      id: "man",
+      isLive: false,
+      lifetime: 13,
+      minimumTime: 4,
+      parsingErrors: ["a", "b"],
+      periods: [{ id: "0" }, { id: "1" }],
+      presentationLiveGap: 18,
+      suggestedPresentationDelay: 99,
+      timeShiftBufferDepth: 2,
+      transportType: "foobar",
+      uris: ["url1", "url2"],
+    };
+
+    const logSpy = jest.spyOn(log, "warn").mockImplementation(jest.fn());
+    const fakePeriod = jest.fn((period) => {
+      return { ...period, id: "foo" + period.id, parsingErrors: [period.id] };
+    });
+    const fakeUpdatePeriodInPlace = jest.fn((oldPeriod, newPeriod) => {
+      Object.keys(oldPeriod).map(key => {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Consider using "forEach" instead of "map" as its return value is not being used here. ([typescript:S2201](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS2201))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=439&issues=AWgpECqo_MPaBDn0IVT0&open=AWgpECqo_MPaBDn0IVT0)

[ISSUE_KEY]: AWgpECqo_MPaBDn0IVT0